crates/autopilot/src/limit_orders/quoter.rs

```diff
@@ -58,10 +58,23 @@ impl LimitOrderQuoter {
 
     /// Returns whether it is likely that there is no more work.
     async fn update(&self) -> Result<bool> {
-        let order_specs = self
+        let orders = self
             .database
-            .order_specs_with_outdated_fees(self.limit_order_age, self.parallelism)
+            .open_limit_orders(self.limit_order_age)
             .await?;
+
+        let orders = match self.strategies.contains(&QuotingStrategy::SkipUnfunded) {
+            true => orders_with_sufficient_balance(&*self.balance_fetcher, orders).await,
+            false => orders,
+        };
+
+        let order_specs = orders
+            .into_iter()
+            .map(order_spec_from)
+            .unique()
+            .take(self.parallelism)
```

Comment:
Related to discussion below. If we are fetching all orders from the DB anyway, then it doesn't make much sense to only `take self.parallelism` here.

The implementation that I originally wrote was designed to take as few orders from the DB as needed to have small individual update loops. This has a benefit over fetching all then updating all, because a single update of all orders can take a long time and by the end some orders might no longer be needed to be updated at all.

Since then we've had the change to group orders (used to happen in DB, now happens here), and now to also filter out orders without enough balance.

Now it seems more wasteful to always get all db orders, and all their balance to then only take ~5 order specs per update. I don't know what the right value is but we can do something like `5 * self.parallelism` or we can update as many order specs as possible at the current parallelism level, until a time limit of 60 seconds is exceeded.

Reply:
I think that is a good proposal. I think we need to stream the price estimates like we used to do in `solvable_orders` then.

Reply:
It's not obvious to me how using a timeout would be better than simply picking more orders to update and wait until they are done. This is trivial to implement and we can't "waste" any requests by starting one and dropping it before we get a response.

Reply:
The reason I suggest a timeout is

> This has a benefit over fetching all then updating all, because a single update of all orders can take a long time and by the end some orders might no longer be needed to be updated at all.

It is a minor optimization.

Reply:
For now, should we specify `self.update_batch_size`, which gets parallelized by `self.parallelism` as @vkgnosis suggested?